upp/stages/hist.py

```diff
@@ -50,47 +51,24 @@ def pdf(self) -> np.array:
         with h5py.File(self.path) as f:
             return f["pdf"][:]
 
-
-"""
-    @functools.cached_property
-    def upscaled_pdf(self):
-        upscale = 1
+    def upscaled_pdf(self, upscl=2):
         # get bins
+        # upscl must be integer
```

Comment:
Let's reflect this by adding type hints to the argument rather than a comment here

Reply:
I have added type hints and the docstrings to all new functions